addons/sale_gift_card/models/sale_order.py

```diff
@@ -50,6 +50,12 @@ def _pay_with_gift_card(self, gift_card):
             })
         return error
 
+    def _compute_amount_total_without_delivery(self):
+        self.ensure_one()
+        # Add back 'payment' rewards from the total without delivery, they should count towards the delivery price goal.
+        lines = self.order_line.filtered(lambda l: l.gift_card_id)
+        return super()._compute_amount_total_without_delivery() + sum(lines.mapped('price_unit'))
```

Comment:
Gift card lines have negative amounts so we actually need to subtract that amount from the total.
```suggestion
        return super()._compute_amount_total_without_delivery() - sum(lines.mapped('price_unit'))
```